### PR TITLE
Bump k8s APIs

### PR DESCRIPTION
**Pull request by pavolloffay**
Signed-off-by: Pavol Loffay <p.loffay@gmail.com>

**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -234,23 +234,22 @@ require (
...
 	google.golang.org/genproto v0.0.0-20220616135557-88e70c0c3a90 // indirect
 	gopkg.in/ini.v1 v1.66.4 // indirect
-	k8s.io/utils v0.0.0-20210930125809-cb0fa318a74b // indirect
 )
 
 replace (
 	github.com/bradfitz/gomemcache => github.com/themihai/gomemcache v0.0.0-20180902122335-24332e2d58ab
-	k8s.io/api => k8s.io/api v0.20.4
-	k8s.io/client-go => k8s.io/client-go v0.20.4
+	k8s.io/api => k8s.io/api v0.25.0
```
Comment (pavolloffay) [inline go.mod:251]: @joe-elliott when it was removed it failed on 

```bash
github.com/grafana/tempo/modules/generator/storage imports
	github.com/prometheus/prometheus/config tested by
	github.com/prometheus/prometheus/config.test imports
	github.com/prometheus/prometheus/discovery/kubernetes imports
	k8s.io/client-go/kubernetes imports
	k8s.io/client-go/discovery imports
	github.com/googleapis/gnostic/OpenAPIv2: module github.com/googleapis/gnostic@latest found (v0.6.9), but does not contain package github.com/googleapis/gnostic/OpenAPIv2
github.com/grafana/tempo/modules/generator/storage imports
	github.com/prometheus/prometheus/config tested by
	github.com/prometheus/prometheus/config.test imports
	github.com/prometheus/prometheus/discovery/kubernetes imports
	k8s.io/client-go/kubernetes imports
	k8s.io/client-go/kubernetes/typed/auditregistration/v1alpha1 imports
	k8s.io/api/auditregistration/v1alpha1: module k8s.io/api@latest found (v0.25.1), but does not contain package k8s.io/api/auditregistration/v1alpha1
github.com/grafana/tempo/modules/generator/storage imports
	github.com/prometheus/prometheus/config tested by
	github.com/prometheus/prometheus/config.test imports
	github.com/prometheus/prometheus/discovery/kubernetes imports
	k8s.io/client-go/kubernetes imports
	k8s.io/client-go/kubernetes/typed/batch/v2alpha1 imports
	k8s.io/api/batch/v2alpha1: module k8s.io/api@latest found (v0.25.1), but does not contain package k8s.io/api/batch/v2alpha1
github.com/grafana/tempo/modules/generator/storage imports
	github.com/prometheus/prometheus/config tested by
	github.com/prometheus/prometheus/config.test imports
	github.com/prometheus/prometheus/discovery/kubernetes imports
	k8s.io/client-go/kubernetes imports
	k8s.io/client-go/kubernetes/typed/settings/v1alpha1 imports
	k8s.io/api/settings/v1alpha1: module k8s.io/api@latest found (v0.25.1), but does not contain package k8s.io/api/settings/v1alpha1

```

**Comment by joe-elliott** [comment]
Thanks for the update. You will likely need to run `make update-mod` to update all vendoring.

**Comment by pavolloffay** [comment]
PR is updated

**Review by joe-elliott — APPROVED** [review]
Thanks!

**Pull request merged**